python/pyarrow/_substrait.pyx

```diff
@@ -124,7 +124,13 @@ def run_query(plan, table_provider=None):
         function[CNamedTableProvider] c_named_table_provider
         CConversionOptions c_conversion_options
 
-    c_buf_plan = pyarrow_unwrap_buffer(plan)
+    if isinstance(plan, bytes):
+        c_buf_plan = pyarrow_unwrap_buffer(py_buffer(plan))
+    elif isinstance(plan, Buffer):
+        c_buf_plan = pyarrow_unwrap_buffer(plan)
+    else:
+        raise TypeError(
+            f"Expected '{Buffer}' or bytes, got '{type(plan)}'")
```

Comment:
```suggestion
            f"Expected pyarrow.Buffer or bytes, got '{type(plan)}'")
```

(automatically printing it will give `"<class 'pyarrow.lib.Buffer'>"`, which is more verbose)

Reply:
ya, I initially had that, but thought if it's ever changed or relocated this would be outdated. I agree it's prettier though.

Reply:
> if it's ever changed or relocated

I think that's quite unlikely